### PR TITLE
Throw an exception when source paths don't exist or no source files are found

### PR DESCRIPTION
**Pull request by johannescoetzee**
__Motivation:__ We've run into issues in the past where invalid source directories were given, but `*2cpg` failed silently. These silent failures are difficult to debug (especially with customer submissions) and are most likely due to user error. This PR adds 2 failure modes when finding source files:
* __The input path doesn't exist or isn't readable:__ this is most likely due to a typo in the input path. This should definitely be an exception that terminates execution since this does not match user intent.
* ~__No source files could be found at the given input paths:__ This could be an acceptable error (for example when finding config files), but in general will not be. The current implementation always throws an exception (which can then be handled by the caller if an empty source list is acceptable), but adding an option to allow an empty sources list (with the default still being to not allow them) is potentially a better option.~

### CODE REVIEW

**Comment by johannescoetzee** [comment]
Another option is to have the `determine` methods return `Try[List[String]]`. This is the better functional approach and will force all callers to handle the success/failure explicitly. This does come at the cost of being a more obtrusive change, however.

File: joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/SourceFiles.scala
Comment (ml86) [inline joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/SourceFiles.scala:105]: Please reuse existing Java exceptions if possible. `NoSuchFileException` should be fine here.

**Review by ml86 — APPROVED** [review]
I like it, but this has the potential to break stuff if people used this for possibly non existing input path in the past. I think we should fix those cases but please wait for @fabsx00 approval to merge.

**Review by fabsx00 — APPROVED** [review]

**Pull request merged**
